lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -335,6 +335,20 @@ static SILValue lookThroughSingleElementStructInsts(SILValue value) {
   return value;
 }
 
+/// If the specified struct has a single stored field, return it.  Otherwise
+/// return null.
+static VarDecl *getFieldIfSingleFieldStruct(StructDecl *decl) {
```

Comment:
You might want to refactor and reuse irgen::getSingletonAggregateFieldType() in GenType.cpp

Reply:
Thanks for the pointer.  SILOptimizer is not supposed to depend on IRGen though, and getSingletonAggregateFieldType is working in terms of resilience expansions.  This code needs to align with the SIL version of things.